packages/git/src/browser/diff/git-diff-model.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import URI from '@theia/core/lib/common/uri';
+
+/**
+ * Options for further refining the `git log` commands.
+ */
+export interface GitLogOptions {
+
+    /**
+     * The uri of a file to run the `git log` command. If not set, the log for all files in the given range are fetched.
+     */
+    readonly fileUri?: string;
+
+    /**
+     * The name of the branch to run the `git log` command. If not specified, then the currently active branch will be used.
+     */
+    readonly branch?: string;
```

Comment:
Is the branch really needed here?

Reply:
Well, ideally, we could move the options to the core `git` module, where it can be an optional parameter. But no, it is not really needed for the diff.